monkestation/code/datums/mutations/active.dm

```diff
@@ -0,0 +1,271 @@
+		REMOVE_TRAIT(owner, TRAIT_STABILIZED_EATER, GENETIC_MUTATION)
+
+/datum/mutation/human/consumption/modify()
+	. = ..()
+	if(!.)
+		return
+
+	var/datum/action/cooldown/spell/pointed/consumption/to_modify = .
+	to_modify.healing_multiplier *= GET_MUTATION_POWER(src)
+	if(GET_MUTATION_SYNCHRONIZER(src) < 1)
+		ADD_TRAIT(owner, TRAIT_STABILIZED_EATER, GENETIC_MUTATION)
+
+/datum/action/cooldown/spell/pointed/consumption
+	name = "Eat Matter"
+	desc = "Eat just about anything!"
+	button_icon = 'icons/mob/actions/actions_animal.dmi'
+	button_icon_state = "regurgitate"
+
+	cooldown_time = 1 MINUTE
+	check_flags = AB_CHECK_CONSCIOUS
+	spell_requirements = NONE
+	antimagic_flags = NONE
+	cast_range = 1
+	aim_assist = FALSE
+
+	var/healing_multiplier = 1
+
+/datum/action/cooldown/spell/pointed/consumption/IsAvailable(feedback = FALSE)
+	if(owner)
+		var/mob/living/carbon/human = owner
+		if(istype(owner) && human.is_mouth_covered())
+			if(feedback)
+				owner.balloon_alert(owner, "mouth blocked!")
+			return FALSE
+
+	return ..()
+
+/datum/action/cooldown/spell/pointed/consumption/before_cast(atom/cast_on)
+	. = ..()
+	if(. & SPELL_CANCEL_CAST)
+		return
+
+	if(isitem(cast_on))
+		var/obj/item/item = cast_on
+		if(item.item_flags & ABSTRACT)
+			cast_on.balloon_alert(owner, "... no.")
+			return . | SPELL_CANCEL_CAST
+
+	if(iseffect(cast_on))
+		cast_on.balloon_alert(owner, "what even is this?")
+		return . | SPELL_CANCEL_CAST
+
+	if(!isturf(owner.loc))
+		cast_on.balloon_alert(owner, "can't eat here!")
+		return . | SPELL_CANCEL_CAST
+
+	if(is_type_in_typecache(cast_on, GLOB.oilfry_blacklisted_items))
+		cast_on.balloon_alert(owner, "a nuclear bomb looks tastier than this.")
+		return . | SPELL_CANCEL_CAST
+
+	if(istype(cast_on, /obj/machinery/nuclearbomb))
+		cast_on.balloon_alert(owner, "a literal nuclear bomb is not great for digestion.")
+		return . | SPELL_CANCEL_CAST
+
+	if(ishuman(cast_on))
+		var/mob/living/carbon/human/human_target = cast_on
+		if(owner.zone_selected == BODY_ZONE_PRECISE_GROIN)
+			var/message = pick(list("... You wouldn't.", "nope.", "better not.", "not a good idea."))
+			cast_on.balloon_alert(owner, message)
+			return . | SPELL_CANCEL_CAST
+
+		if(owner.zone_selected == BODY_ZONE_CHEST)
+			cast_on.balloon_alert(owner, "too big!")
+			return . | SPELL_CANCEL_CAST
+
+		var/obj/item/bodypart/limb = human_target.get_bodypart(owner.zone_selected)
+		if(!limb)
+			cast_on.balloon_alert(owner, "nothing there!") // Is that a reference to project m-
+			return . | SPELL_CANCEL_CAST
+
+		if(owner.zone_selected == BODY_ZONE_HEAD)
+			if(owner.pulling != cast_on)
+				cast_on.balloon_alert(owner, "need grasp!")
+				return . | SPELL_CANCEL_CAST
+
+			if(owner.grab_state != GRAB_KILL)
+				cast_on.balloon_alert(owner, "grasp too loose!")
+				return . | SPELL_CANCEL_CAST
+
+	StartCooldown()
+	return . | SPELL_NO_IMMEDIATE_COOLDOWN
+
+/datum/action/cooldown/spell/pointed/consumption/cast(obj/cast_on)
```

Comment:
PLEASE refactor this to use some sort of signal instead of just a bunch of individual istypes

Reply:
Fair, gonna do dat

Reply:
and done